### ISSUE TITLE
Update Configuration Backup to GitHub page

### ISSUE THREAD

**Issue by frenck** [home-assistant/home-assistant.github.io#6994]
The "Configuration Backup to GitHub" page, is really outdated.

Home Assistant changed a lot, and the .gitignore example is far from complete.
If a user follows this, he definitely will expose secrets/tokens to GitHub.

The page:
<https://www.home-assistant.io/docs/ecosystem/backup/backup_github/#step-6-keeping-your-repository-up-to-date>

e.g.: `ip_bans.yaml`

**Comment by greybax** [comment]
I would work on it, but I'm not sure what should be done here..

**Comment by klaasnicolaas** [comment]
Added some extra things to the .gitignore example 😉

**Comment by frenck** [comment]
I've seen, and @fabaff approved.
But it doesn't make sense to me. For example:
https://github.com/home-assistant/home-assistant.io/pull/7072/files#diff-77ee43d21a06e4ce00e1581c91c1ce04R64

This file is already ignored by other rules in the file, so now a duplicate.

**Comment by klaasnicolaas** [comment]
In that case I will remove that line.

Edit: see also now that the `*` automatically ensures that important files or folders are not exposed to the outside, then there is nothing wrong with the example?

**Comment by frenck** [comment]
Yes there is, `ip_bans.yml` for example.

**Comment by klaasnicolaas** [comment]
Will create a new PR with WIP

**Comment by point-4ward** [comment]
Why don't we just link to this (or pull the page text from it with an attribution)? 

https://www.gitignore.io/api/homeassistant

(edit - that needs updating too, but it's a lot better than the current one) 

What else are you thinking for the whole GitHub page in general, @Frenck ? I'll happily rewrite an ELI5 style backup to GitHub guide in its place but it runs the risk of getting overcomplicated fast for users completely new to the platform, and being too dumbed down for people who are techy.

Give me a clue what sort of thing you were thinking and I'll see if I can pitch something to the right level.

**Comment by point-4ward** [comment]
To clarify...

I use an ssh key and automate all my GitHub stuff from homeassistant itself, I have bash scripts and a maintenance package and .gitignores and .gitattributes and protected branches etc, but it's way too much for for a 'basic' guide. 

But then we are a community of automators, so maybe that's the kind of thing you were looking for? 

Basically, I don't want to spend hours writing something like that if it's too much :wink:

**Comment by frenck** [comment]
@mf-social The approach is more a white-list kinda approach, instead of blacklisting (which is more usual). Nevertheless, it protects the use better against changes in Home Assistant.
